rasa/utils/tensorflow/rasa_layers.py

```diff
@@ -28,9 +29,182 @@
+                    "was run. The training data was changed in a way that resulted in "
+                    "some features not being present in the data anymore. This can "
+                    "happen if you had `LexicalSyntacticFeaturizer` in your pipeline. "
+                    "The pipeline cannot support incremental training in this setting."
+                    " We recommend you to retrain the model from scratch."
+                )
+
+    @staticmethod
+    def _replace_dense_for_sparse_layer(
+        layer_to_replace: layers.DenseForSparse,
+        new_sparse_feature_sizes: List[int],
+        old_sparse_feature_sizes: List[int],
+        attribute: Text,
+        feature_type: Text,
+        reg_lambda: float,
+    ) -> layers.DenseForSparse:
+        """Replaces a `DenseForSparse` layer with a new one.
+
+        Replaces an existing `DenseForSparse` layer with a new one
+        in order to adapt it to incremental training.
+
+        Args:
+            layer_to_replace: a `DenseForSparse` layer that is used to create a new one.
+            new_sparse_feature_sizes: sizes of sparse features that will be
+                                      the input of the layer.
+            old_sparse_feature_sizes: sizes of sparse features that used to be
+                                      the input of the layer.
+            attribute: an attribute of the data fed to the layer.
+            feature_type: a feature type of the data fed to the layer.
+            reg_lambda: regularization constant.
+
+        Returns:
+            New `DenseForSparse` layer.
+        """
+        kernel = layer_to_replace.get_kernel().numpy()
+        bias = layer_to_replace.get_bias()
+        use_bias = False if bias is None else True
+        if use_bias:
+            bias = bias.numpy()
+        units = layer_to_replace.get_units()
+        # split kernel by feature sizes to update the layer accordingly
+        kernel_splits = []
+        splitting_index = 0
+        for size in old_sparse_feature_sizes:
+            kernel_splits.append(kernel[splitting_index : splitting_index + size, :])
+            splitting_index += size
+        additional_sizes = [
+            new_size - old_size
+            for new_size, old_size in zip(
+                new_sparse_feature_sizes, old_sparse_feature_sizes
+            )
+        ]
+        std, mean = np.std(kernel), np.mean(kernel)
+        additional_weights = [
+            np.random.normal(mean, std, size=(num_rows, units)).astype(np.float32)
+            for num_rows in additional_sizes
+        ]
+        merged_weights = [
+            np.vstack((existing, new))
+            for existing, new in zip(kernel_splits, additional_weights)
+        ]
+        # stack each merged weight to form a new weight tensor
+        new_weights = np.vstack(merged_weights)
+        kernel_init = tf.constant_initializer(new_weights)
+        bias_init = tf.constant_initializer(bias)
```

Comment:
Does `tf.constant_initializer` accept `None` as value? Because `bias` can potentially be `None`.